drivers/ieee802154/ieee802154_stm32wba.c

```diff
@@ -0,0 +1,1051 @@
+		LOG_DBG("Setting MIN_CSMA_BE: %u", config->min_csma_be);
+		st_ral_set_min_csma_be(config->min_csma_be);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_MAX_CSMA_BE:
+		LOG_DBG("Setting MAX_CSMA_BE: %u", config->max_csma_be);
+		st_ral_set_max_csma_be(config->max_csma_be);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_MAX_CSMA_BACKOFF:
+		LOG_DBG("Setting MAX_CSMA_BACKOFF: %u", config->max_csma_backoff);
+		st_ral_set_max_csma_backoff(config->max_csma_backoff);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_IMPLICIT_BROADCAST:
+		LOG_DBG("Setting IMPLICIT_BROADCAST: %u", config->impl_brdcast);
+		st_ral_set_implicitbroadcast(config->impl_brdcast);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_ANTENNA_DIV:
+		LOG_DBG("Setting ANTENNA_DIV: %u", config->ant_div);
+		if (st_ral_set_ant_div_enable(config->ant_div)) {
+			return -EIO;
+		}
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_RADIO_RESET:
+		LOG_DBG("Setting RADIO_RESET");
+		if (st_ral_radio_reset()) {
+			return -EIO;
+		}
+		break;
+
+	default:
+		LOG_ERR("Unsupported configuration type: %d", type);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int stm32wba_attr_get_extended(enum ieee802154_stm32wba_attr attr,
+				      const struct ieee802154_stm32wba_attr_value *value)
+{
+	switch ((uint32_t)attr) {
+	case IEEE802154_STM32WBA_ATTR_CCA_THRESHOLD:
+		LOG_DBG("Getting CCA_THRESHOLD attribute");
+		static int8_t l_cca_thr;
+
+		if (st_ral_get_cca_energy_detect_threshold(&l_cca_thr)) {
+			return -ENOENT;
+		}
+		((struct ieee802154_stm32wba_attr_value *)value)->cca_thr = &l_cca_thr;
+		break;
+
+	case IEEE802154_STM32WBA_ATTR_IEEE_EUI64:
+		LOG_DBG("Getting IEEE_EUI64 attribute");
+		uint8_t l_eui64[8];
+
+		stm32wba_get_eui64(l_eui64);
+		memcpy(((struct ieee802154_stm32wba_attr_value *)value)->eui64, l_eui64,
+				sizeof(l_eui64));
+		break;
+
+	case IEEE802154_STM32WBA_ATTR_TX_POWER:
+		LOG_DBG("Getting TX_POWER attribute");
+		static uint8_t l_tx_power;
+
+		if (st_ral_tx_power_get(&l_tx_power)) {
+			return -ENOENT;
+		}
+		((struct ieee802154_stm32wba_attr_value *)value)->tx_power = &l_tx_power;
+		break;
+
+	case IEEE802154_STM32WBA_ATTR_RAND_NUM:
+		LOG_DBG("Getting RAND_NUM attribute");
+		static uint8_t l_rand_num;
+
+		if (st_ral_mac_gen_rnd_num(&l_rand_num, 1, true)) {
+			return -ENOENT;
+		}
+		((struct ieee802154_stm32wba_attr_value *)value)->rand_num = &l_rand_num;
+		break;
+
+	default:
+		LOG_ERR("Unsupported attribute: %u", (uint32_t)attr);
+		return -ENOENT;
+	}
+
+	return 0;
+}
+#endif
+
+/* Radio device API */
+
+static enum ieee802154_hw_caps stm32wba_get_capabilities(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return IEEE802154_HW_ENERGY_SCAN |
+		IEEE802154_HW_FCS |
+		IEEE802154_HW_FILTER |
+		IEEE802154_HW_PROMISC |
+#if CONFIG_IEEE802154_STM32WBA_CSMA_CA_ENABLED
+		IEEE802154_HW_CSMA |
+#endif
+		IEEE802154_HW_TX_RX_ACK |
+		IEEE802154_HW_RETRANSMISSION |
+		IEEE802154_HW_RX_TX_ACK |
+#if (CONFIG_MAC_CSL_TRANSMITTER_ENABLE == 1)
```

Comment:
I can't see this configuration option defined anywhere. Neither `CONFIG_MAC_CSL_RECEIVER_ENABLE` used 2 lines below.

Note that for config options, the related `CONFIG_xxx` macro is defined when the option is enabled and not defined when the option is disabled. Therefore here you should use `#ifdef`.